### PR TITLE
test-bot only test the runtime dependencies

### LINKED ISSUE #36154: test-bot should only test the runtime dependencies
So I just update `cmake` to 3.1.1, and CI bot spends lots of time to test all of formulae which depend on `cmake`.

We shouldn't do this. Instead, we should only test the formulae which depend on `cmake` on runtime as most of them only use `cmake` in building time.

### PR DESCRIPTION
**Pull request by xu-cheng**
Close #36154

### CODE REVIEW

File: Library/Homebrew/cmd/uses.rb [hunk 1]
```diff
@@ -11,15 +11,26 @@ def uses
...
+    end
 
     uses = formulae.select do |f|
       used_formulae.all? do |ff|
         begin
           if recursive
-            f.recursive_dependencies.any? { |dep| dep.to_formula.name == ff.name } ||
+            f.recursive_dependencies.reject do |dep|
+              ignore_deps.any? { |type| dep.send(type) }
+            end.any? { |dep| dep.to_formula.name == ff.name } ||
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/uses.rb]: The `end.any?` I find a bit weird; would be good to combine or split this.

File: Library/Homebrew/cmd/uses.rb [hunk 2]
```diff
@@ -11,15 +11,26 @@ def uses
     used_formulae = ARGV.formulae
     formulae = (ARGV.include? "--installed") ? Formula.installed : Formula
     recursive = ARGV.flag? "--recursive"
+    if ARGV.flag? "--runtime-only"
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/uses.rb]: `--ignore-build`?

File: Library/Homebrew/cmd/uses.rb [hunk 3]
```diff
@@ -11,15 +11,26 @@ def uses
...
+      ignore_deps = %w[build?]
+    elsif ARGV.flag? "--bot-only"
+      ignore_deps = %w[build? optional?]
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/uses.rb]: `--ignore-optional` to ignore just optional or `--default` or something?

**Comment by mikemcquaid** [comment]
Nice work here :+1:

**Comment by xu-cheng** [comment]
Updated.

File: Library/Homebrew/manpages/brew.1.md [hunk 4]
```diff
@@ -407,6 +407,10 @@ Note that these flags should only appear after a command.
 
     If `--installed` is passed, only list installed formulae.
 
+    By default, `uses` shows all formulae that specify <formulae> as a dependency.
+    To skip the `:build` type dependencies, pass `--skip-build`. Similar, pass
```
Comment (mikemcquaid) [inline Library/Homebrew/manpages/brew.1.md]: `Similar` -> `Similarly`

**Comment by mikemcquaid** [comment]
Final nit but otherwise looks great, fantastic work here @xu-cheng!

**Comment by xu-cheng** [comment]
Fixed. :tada:

**Pull request closed without merging**